jdaviz/configs/default/plugins/model_fitting/model_fitting.py

```diff
@@ -720,7 +722,13 @@ def _fit_model_to_cube(self, add_data):
             return
 
         # Get the primary data component
-        spec = data.get_object(Spectrum1D, statistic=None)
+        if not self.spatial_subset_selected.startswith("Entire"):
+            # get_subset_object supports masking but get_object does not
+            spec = data.get_subset_object(
+                self.spatial_subset_selected, cls=Spectrum1D, statistic=None
+            )
+        else:
+            spec = data.get_object(Spectrum1D, statistic=None)
```

Comment:
Does this support either the subset name or cls as the first positional argument?

```suggestion
            spec = data.get_object(cls=Spectrum1D, statistic=None)
```

Reply:
There's a helpful demo [here](https://glue-astronomy.readthedocs.io/en/latest/translators.html#datasets-and-subsets): `get_object` takes `cls`, but `get_subset_object` takes the subset as the first argument, then `cls`.